kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -251,24 +348,32 @@ private BigQuery getBigQuery() {
     if (testBigQuery != null) {
       return testBigQuery;
     }
+    return bigQuery.updateAndGet(bq -> bq != null ? bq : newBigQuery());
```

Comment:
Are both `BigQuery` and `SchemaManager` objects thread-safe? Looks like they were not being shared across threads before whereas they may be now. Why share them instead of having them in `ThreadLocal`s and still have just one for each thread?

Reply:
I think they both are used concurrently right now since they get given (directly or indirectly) to `TableWriterBuilder` instances when are then each run on their own threads.

There are also some changes here to the `SchemaManager` class that should make help alleviate some performance problems caused by concurrent usage by optimizing away redundant operations. They don't make it more thread-safe since it already appears to be, but they do make it friendlier to concurrent use and help avoid unnecessary DDL requests to BigQuery.